pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go

```diff
@@ -358,3 +377,50 @@ func (p *partitionedStreamSubscription) Events() <-chan streamingccl.Event {
 func (p *partitionedStreamSubscription) Err() error {
 	return p.err
 }
+
+func URIWithTLSCertsRemoved(remote *url.URL) *url.URL {
```

Comment:
A pattern which I believe @adityamaru has used in a few places is to make this a single method that "consumes" some of the parameters and returns pair of the config struct built from them and the URI now without them. Not sure if that'd fit here but I sorta like its conceptual tidiness versus having to mentally pair two separate methods.

Reply:
Cool, I was going back and forth on this.  I pushed _something_ like this. It isn't as clean as I might like, but I wanted all of the certificate processing errors to happen up front.